apps/trading/components/asset-activity/asset-activity.tsx

```diff
@@ -0,0 +1,355 @@
+import { useMemo, useState } from 'react';
+import compact from 'lodash/compact';
+
+import { AgGrid, COL_DEFS, SetFilter } from '@vegaprotocol/datagrid';
+import {
+  type DepositFieldsFragment,
+  useDeposits,
+} from '@vegaprotocol/deposits';
+import {
+  type WithdrawalFieldsFragment,
+  useWithdrawals,
+} from '@vegaprotocol/withdraws';
+import { useVegaWallet } from '@vegaprotocol/wallet-react';
+import {
+  useAssetDetailsDialogStore,
+  type AssetFieldsFragment,
+} from '@vegaprotocol/assets';
+import {
+  type TransferFieldsFragment,
+  useTransfers,
+} from '@vegaprotocol/accounts';
+import {
+  addDecimalsFormatNumber,
+  getDateTimeFormat,
+  toBigNum,
+} from '@vegaprotocol/utils';
+import {
+  DepositStatus,
+  TransferStatus,
+  TransferStatusMapping,
+  WithdrawalStatus,
+} from '@vegaprotocol/types';
+
+import { useT } from '../../lib/use-t';
+import { DepositStatusCell } from './deposit-status-cell';
+import { WithdrawalStatusCell } from './withdrawal-status-cell';
+import { DepositToFromCell } from './deposit-to-from-cell';
+import { WithdrawalToFromCell } from './withdrawal-to-form-cell';
+import { TransferToFromCell } from './transfer-to-from-cell';
+import { Pagination } from '@vegaprotocol/datagrid';
+
+export interface RowBase {
+  asset: AssetFieldsFragment | undefined;
+  amount: string;
+  bAmount: BigNumber;
+  createdTimestamp: Date;
+}
+
+export interface RowDeposit extends RowBase {
+  type: 'Deposit';
+  detail: DepositFieldsFragment;
+}
+
+export interface RowWithdrawal extends RowBase {
+  type: 'Withdrawal';
+  detail: WithdrawalFieldsFragment;
+}
+
+export interface RowTransfer extends RowBase {
+  type: 'Transfer';
+  detail: TransferFieldsFragment;
+}
+
+export type Row = RowDeposit | RowWithdrawal | RowTransfer;
+
+const PAGE_SIZE = 1000;
+
+/** Used for making all status between deposits/withdraws/transfers consistent */
+export const StatusSet = {
+  Pending: 'Pending',
+  Failed: 'Failed',
+  Complete: 'Complete',
+  Cancelled: 'Cancelled',
+  Stopped: 'Stopped',
+} as const;
+
+export const AssetActivity = () => {
+  const { pubKey } = useVegaWallet();
+  const [first, setFirst] = useState(PAGE_SIZE);
+
+  const { data: deposits, pageInfo: depositsPageInfo } = useDeposits({
+    pubKey,
+    pagination: { first },
+  });
+  const { data: withdrawals, pageInfo: withdrawalsPageInfo } = useWithdrawals({
+    pubKey,
+    pagination: { first },
+  });
+  const { data: transfers, pageInfo: transfersPageInfo } = useTransfers({
+    pubKey,
+    pagination: { first },
+  });
+
+  const hasNextPage = [
+    depositsPageInfo?.hasNextPage,
+    withdrawalsPageInfo?.hasNextPage,
+    transfersPageInfo?.hasNextPage,
+  ].some((hasNextPage) => hasNextPage);
+
+  const rowData = useMemo(() => {
+    return compact(
+      [...(deposits || []), ...(withdrawals || []), ...(transfers || [])].map(
+        normalizeItem
+      )
+    );
+  }, [deposits, withdrawals, transfers]);
```

Comment:
**praise (non-blocking)**:  this is nicely encapsulated for the amount of data munging it's doing